In [1]:
import pandas as pd

df = pd.read_csv('filtered/tweets.csv')
pd.set_option('display.max_columns', None)

ParserError: Error tokenizing data. C error: Expected 83 fields in line 3, saw 84


In [3]:
df.shape

(15067, 184)

In [4]:
df.head()

created_at                   id               id_str  \
0  Mon Feb 10 07:05:27 +0000 2020  1226764098374950912  1226764098374950912   
1  Mon Feb 10 07:11:45 +0000 2020  1226765683100397569  1226765683100397569   
2  Mon Feb 10 07:13:10 +0000 2020  1226766037611356160  1226766037611356160   
3  Mon Feb 10 07:18:16 +0000 2020  1226767322477514752  1226767322477514752   
4  Mon Feb 10 07:23:07 +0000 2020  1226768543754133504  1226768543754133504   

                                           full_text  truncated  \
0  #OBRAS nivel AMARILLO en CARRETERA #EI500 (pk ...      False   
1  Welcome to corona (@ İstanbul Havalimanı - @ig...      False   
2  Indonesia sakjozee!! 😅\n.\n.\n.\nWabah virus c...      False   
3  #maskman #fakesoldout @ Sanya, Hainan, China h...      False   
4  とはいえニュースは流れてきている。主要メディアのSNS見てる程度ですが https://t....      False   

  display_text_range                                             source  \
0           [0, 139]  <a href="http://www.dgt.es" rel="nofollow">D.G...   
1           [0, 102]  <a href="http://foursquare.com" rel="nofollow"...   
2           [0, 235]  <a href="http://instagram.com" rel="nofollow">...   
3            [0, 68]  <a href="http://instagram.com" rel="nofollow">...   
4            [0, 35]  <a href="https://onloft.com/tweetlogix" rel="n...   

   in_reply_to_status_id  in_reply_to_status_id_str  in_reply_to_user_id  \
0                    NaN                        NaN                  NaN   
1                    NaN                        NaN                  NaN   
2                    NaN                        NaN                  NaN   
3                    NaN                        NaN                  NaN   
4                    NaN                        NaN                  NaN   

   in_reply_to_user_id_str in_reply_to_screen_name  contributors  \
0                      NaN                     NaN           NaN   
1                      NaN                     NaN           NaN   
2                      NaN                     NaN           NaN   
3                      NaN                     NaN           NaN   
4                      NaN                     NaN           NaN   

   is_quote_status  retweet_count  favorite_count  favorited  retweeted  \
0            False              0               0      False      False   
1            False              0               0      False      False   
2            False              0               1      False      False   
3            False              0               0      False      False   
4             True              0               0      False      False   

  possibly_sensitive lang                                  entities.hashtags  \
0              False   ca  [{'text': 'OBRAS', 'indices': [0, 6]}, {'text'...   
1              False   tr                                                 []   
2              False   in                                                 []   
3               True   hi  [{'text': 'maskman', 'indices': [0, 8]}, {'tex...   
4              False   ja                                                 []   

  entities.symbols                             entities.user_mentions  \
0               []                                                 []   
1               []  [{'screen_name': 'igairport', 'name': 'İGA Ist...   
2               []                                                 []   
3               []                                                 []   
4               []                                                 []   

                                       entities.urls              user.id  \
0  [{'url': 'https://t.co/jaUPwBDsJS', 'expanded_...            974022020   
1  [{'url': 'https://t.co/x7zhj8EkBd', 'expanded_...           1848965390   
2  [{'url': 'https://t.co/6sDYCZI5N2', 'expanded_...           1376369066   
3  [{'url': 'https://t.co/Nbo9jzXiBU', 'expanded_...  1147390009151873024   
4  [{'url': 'https://t.co/g8Mz5lA8vc', 'expanded_...             2464272

In [28]:
df['user.id'].value_counts()[0:19]

853007708814954496     370
1316862031             365
208056970              350
191092262              347
46717886               179
73672052               113
185789961              100
229334332               99
157819685               96
1043099688              87
138418838               83
305030732               56
1043310177490034688     55
340422281               53
394421259               47
127943720               46
149585330               44
15392221                41
462476619               35
Name: user.id, dtype: int64

with just about 15k tweets, only the top 7 of the most frequent users have 100 tweets or over. That shows a pretty good distribtuion of different users.

user counts:
370
365
350
347
179
113
100
...

In [8]:
len(df['user.id'].unique())

8771

there are 8771 unique users in the sample data

In [18]:
df['geo.coordinates'].value_counts()[0:19]

[52.232222, 21.008333]          365
[45.75975883, -73.56725453]     277
[31.2, 121.5]                   267
[8.96389499, 7.42936372]        179
[39.9289, 116.388]              173
[37.7929728, -122.3958528]      165
[35.0, 103.0]                   138
[30.5833, 114.267]              116
[55.852472, -4.309626]          113
[45.63697547, -73.56726563]     100
[18.4596876, -69.9485084]        99
[33.88, -117.58]                 96
[51.50711486, -0.12731805]       95
[40.01709786, 116.52099609]      94
[37.7485909, -122.4190191]       86
[23.1167, 113.25]                82
[30.52502419, 114.35014953]      75
[37.71235651, -122.40294686]     72
[40.7142, -74.0064]              70
Name: geo.coordinates, dtype: int64

not a whole lot of repeats for the geo coordinates, very similar, but lower, numbers to user.id. This could suggest that most people are tweeting from a certain location, ie home.

In [11]:
df['user.verified'].value_counts()

False    14631
True       436
Name: user.verified, dtype: int64

only 436 tweets are from a twitter verified user

In [12]:
df['user.geo_enabled'].value_counts()

True     15057
False       10
Name: user.geo_enabled, dtype: int64

as expected, almost all but 10 tweets have the geo location enabled, they were filtered by grabbing the geo locations.

In [14]:
df['place.country_code'].value_counts()[0:19]

US    3770
CN    2318
GB     968
CA     787
IT     631
ID     538
PL     373
NG     366
BR     357
ES     301
TH     296
IN     291
PH     276
MX     275
JP     203
AU     187
DO     179
MY     176
DE     159
Name: place.country_code, dtype: int64

number of tweets per country:
US | 3770
CN | 2318
GB | 968
CA | 787
IT | 631
...

In [10]:
df['quoted_status_id'].value_counts().sum()

173

This tells us that only 173 tweets are quoted

In [12]:
df['full_text'].value_counts()

上海/上海\nShanghai/China at 7 evening                                                                                                                                                                                                                                                 16
#aljazeera Coronavirus: Which countries... https://t.co/DhZLO08E9b                                                                                                                                                                                                                  8
#OBRAS nivel AMARILLO en CARRETERA #EI500 (pk 0 al 11.845 ambos sent.) SANTA AGNES DE CORONA #DGTBalears,Illes #DGT https://t.co/jaUPwBDsJS                                                                                                                                         6
Quotes - She's so fake. If you look behind her neck, I bet it says "Made in China". \n\nLOUDRE TrueLoveOnIWantASAP                                                    

The top fully repeated tweet is only repeated 16 times, this means there is very little repitition in the data set for repeating text

In [30]:
df['retweeted'].value_counts()

False    15067
Name: retweeted, dtype: int64

None of the tweets are retweeted